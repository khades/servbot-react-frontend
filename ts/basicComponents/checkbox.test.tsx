import * as React from "react";
import { create } from "react-test-renderer";
import Checkbox, { ICheckboxProps } from "./checkbox";

describe("Checkbox", () => {
    it("should render and have no errors", () => {
        let startingValue: boolean = false;
        const props: ICheckboxProps = {
            id: "LABEL",
            label: "LABEL",
            setValue: (value: boolean) => { startingValue = value; },
            value: startingValue,
        };
        const checkbox = create(
            <Checkbox {...props} />,
        );
        expect(checkbox.toJSON()).toMatchSnapshot();

        expect(
            checkbox.root.findAll((el) => el.type === "input" && el.props.type === "checkbox").length,
        ).toBeGreaterThan(0);
        expect(checkbox.root.findAll(
            (el) => el.type === "label" && el.children && el.children[0] === "LABEL").length).toBeGreaterThan(0);

        expect(checkbox.root.findAll((el) => el.props.className === "checkbox__errors").length).toBe(0);
        expect(checkbox.root.findAll((el) => el.props.className === "checkbox--error").length).toBe(0);

    });
    it("should render and have errors", () => {
        let startingValue: boolean = true;
        const props: ICheckboxProps = {
            errors: ["1", "2"],
            id: "LABEL",
            label: "LABEL",
            setValue: (value: boolean) => { startingValue = value; },
            value: startingValue,
        };

        const checkbox = create(
            <Checkbox {...props} />,
        );
        expect(checkbox.toJSON()).toMatchSnapshot();
        expect(checkbox.root.findAll(
            (el) => el.props.className === "checkbox__errors").length).toBe(1);
        expect(checkbox.root.findAll(
            (el) => el.props.className === "checkbox checkbox--error").length).toBe(1);
        expect(checkbox.root.findAll(
            (el) => el.type === "label" && el.children && el.children[0] === "LABEL").length).toBeGreaterThan(0);
        expect(checkbox.root.findAll(
            (el) => el.props.className === "checkbox__errors" && el.children && el.children[0] === "1, 2",
        ).length).toBe(1);

    });
    it("should properly change state", () => {
        let startingValue: boolean = false;
        const props: ICheckboxProps = {
            id: "LABEL",
            label: "LABEL",
            setValue: (value: boolean) => { startingValue = value; },
            value: startingValue,
        };
        const checkbox = create(
            <Checkbox {...props} />,
        ).root;
        const input = checkbox.find((el) => el.type === "input" && el.props.type === "checkbox");
        input.props.onClick({ currentTarget: { checked: true } });
        expect(startingValue).toBe(true);
        input.props.onClick({ currentTarget: { checked: false } });
        expect(startingValue).toBe(false);
        input.props.onClick({ currentTarget: { checked: false } });
        expect(startingValue).toBe(false);
    });
});
