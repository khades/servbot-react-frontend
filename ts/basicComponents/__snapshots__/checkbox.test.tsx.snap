// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render and have errors 1`] = `
<div
  className="checkbox checkbox--error"
>
  <input
    checked={true}
    id="LABEL"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="LABEL"
  >
    LABEL
     
  </label>
  <div
    className="checkbox__errors"
  >
    1, 2
  </div>
</div>
`;

exports[`Checkbox should render and have no errors 1`] = `
<div
  className="checkbox"
>
  <input
    checked={false}
    id="LABEL"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="LABEL"
  >
    LABEL
     
  </label>
</div>
`;
